dna/modify.go

```diff
@@ -1,139 +1,185 @@
 package dna
 
 import (
-	"fmt"
-	"github.com/vertgenlab/gonomics/common"
-	"log"
+	"errors"
 )
 
+var (
+	ErrInvalidInterval          = errors.New("deletion interval is not valid")
+	ErrInvalidInsertionPosition = errors.New("insertion position is not valid")
+	ErrUnrecognizedBase         = errors.New("input base was not recognized")
+)
+
+// ToUpper changes the input base to uppercase.
 func ToUpper(b Base) Base {
-	if b == Gap || b == Dot {
-		return b
-	} else if b > N {
-		return b - 5
-	} else {
+	switch b {
+	case LowerA:
+		return A
+	case LowerC:
+		return C
+	case LowerG:
+		return G
+	case LowerT:
+		return T
+	case LowerN:
+		return N
+	default:
 		return b
 	}
 }
 
+// ToLower changes the input base to lowercase.
 func ToLower(b Base) Base {
-	if b == Gap || b == Dot {
-		return b
-	} else if b < a {
-		return b + 5
-	} else {
+	switch b {
+	case A:
+		return LowerA
+	case C:
+		return LowerC
+	case G:
+		return LowerG
+	case T:
+		return LowerT
+	case N:
+		return LowerN
+	default:
 		return b
 	}
 }
 
-// start is closed, end is open, both are zero-based
+// RangeToUpper changes the bases in a set range to uppercase.
+// start is closed, end is open, both are zero-based.
 func RangeToUpper(bases []Base, start int, end int) {
 	for i := start; i < end; i++ {
 		bases[i] = ToUpper(bases[i])
 	}
 }
 
-// start is closed, end is open, both are zero-based
+// RangeToLower changes the bases in a set range to lowercase.
+// start is closed, end is open, both are zero-based.
 func RangeToLower(bases []Base, start int, end int) {
 	for i := start; i < end; i++ {
 		bases[i] = ToLower(bases[i])
 	}
 }
 
+// AllToUpper changes all bases in a sequence to uppercase.
 func AllToUpper(bases []Base) {
 	RangeToUpper(bases, 0, len(bases))
 }
 
+// AllToLower changes all bases in a sequence to lowercase.
 func AllToLower(bases []Base) {
 	RangeToLower(bases, 0, len(bases))
 }
 
-func ComplementSingleBase(b Base) Base {
+// ComplementSingleBase returns the nucleotide complementary to the input base.
```

Comment:
I think we will need additional documentation about what happen when there is an error.  In this case I think they get the starting base/value back.  Maybe we also needs tests for error inputs?